From 74a22ed7560764a06af6b5548ba8c32e001bc782 Mon Sep 17 00:00:00 2001
From: Gabriel Przybysz Gonçalves Júnior <gabriel.przybsyzgoncalvesjunior@windriver.com>
Date: Fri, 24 Oct 2025 08:52:16 -0300
Subject: [PATCH] csi-powermax: add secretcert in templates

This patch consists in add an extra secret containing the https
certificate that is used by the helm chart.

Signed-off-by: Gabriel Przybysz Gonçalves Júnior <gabriel.przybsyzgoncalvesjunior@windriver.com>
---
 charts/csi-powermax/templates/cert-secret.yaml | 10 ++++++++++
 charts/csi-powermax/values.yaml                |  2 +-
 2 files changed, 11 insertions(+), 1 deletion(-)
 create mode 100644 charts/csi-powermax/templates/cert-secret.yaml

diff --git a/charts/csi-powermax/templates/cert-secret.yaml b/charts/csi-powermax/templates/cert-secret.yaml
new file mode 100644
index 0000000..b66d304
--- /dev/null
+++ b/charts/csi-powermax/templates/cert-secret.yaml
@@ -0,0 +1,10 @@
+{{- if .Values.certSecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-certs
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+stringData:
+  {{ .Values.certSecret | toYaml | nindent 2 }}
+{{- end }}
diff --git a/charts/csi-powermax/values.yaml b/charts/csi-powermax/values.yaml
index b1d1c14..575ac89 100644
--- a/charts/csi-powermax/values.yaml
+++ b/charts/csi-powermax/values.yaml
@@ -542,7 +542,7 @@ vSphere:
 proxyAuthzTokens:
   access:
   refresh:
-
+certSecret:
 # Enable this feature only after contact support for additional information
 podmon:
   image: quay.io/dell/container-storage-modules/podmon:v1.14.0
--
2.34.1
